component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/mobileconnect/MobileConnectAuthenticator.java

```diff
@@ -546,48 +616,49 @@ private HttpURLConnection discoveryProcess(String authorizationHeader, String ms
 
         //check whether the msisdn is provided as a parameter
         if (StringUtils.isNotEmpty(msisdn)) {
+
+
             String url = MobileConnectAuthenticatorConstants.DISCOVERY_API_URL + "?" +
-                    MobileConnectAuthenticatorConstants.MOBILE_CONNECT_SESSION_DATA_KEY + "=" + state;
+                    MobileConnectAuthenticatorConstants.MOBILE_CONNECT_DISCOVERY_REDIRECT_URL + "=" +
+                            MobileConnectAuthenticatorConstants.MOBILE_CONNECT_CALLBACK_URL ;
+
+
+            String data = "MSISDN=" + msisdn ;
 
             URL obj = new URL(url);
             HttpURLConnection connection = (HttpURLConnection) obj.openConnection();
 
             connection.setRequestMethod("POST");
-            connection.setRequestProperty(MobileConnectAuthenticatorConstants.MOBILE_CONNECT_DISCOVERY_CONTENT_TYPE,
-                    MobileConnectAuthenticatorConstants.MOBILE_CONNECT_TOKEN_CONTENT_TYPE_VALUE);
             connection.setRequestProperty(MobileConnectAuthenticatorConstants.MOBILE_CONNECT_DISCOVERY_AUTHORIZATION,
                     authorizationHeader);
             connection.setRequestProperty(MobileConnectAuthenticatorConstants.MOBILE_CONNECT_DISCOVERY_ACCEPT,
                     MobileConnectAuthenticatorConstants.MOBILE_CONNECT_DISCOVERY_ACCEPT_VALUE);
             connection.setDoOutput(true);
 
-            //send the data with the connection. remove the plus sign (+) from the msisdn value
-            String data = "MSISDN=%2B" + msisdn.substring(1) + "&Redirect_URL=" +
-                    getCallbackUrl(authenticatorProperties);
-            OutputStreamWriter out = new OutputStreamWriter(connection.getOutputStream(), StandardCharsets.UTF_8);
-            out.write(data);
-            out.close();
+//            send the data with the connection. remove the plus sign (+) from the msisdn value
+
+
+            DataOutputStream outputStream = new DataOutputStream(connection.getOutputStream());
+            outputStream.writeBytes(data);
+            outputStream.close();
 
             return connection;
         }
         //if msisdn is null - not provided by the service provider
         else {
 
-            String url = MobileConnectAuthenticatorConstants.DISCOVERY_API_URL + "?" +
-                    MobileConnectAuthenticatorConstants.MOBILE_CONNECT_DISCOVERY_REDIRECT_URL + "=" +
-                    getCallbackUrl(authenticatorProperties) + "&" +
-                    MobileConnectAuthenticatorConstants.MOBILE_CONNECT_SESSION_DATA_KEY + "=" + state + "&"
-                    + MobileConnectAuthenticatorConstants.MOBILE_CONNECT_MANUAL_SELECTION + "=" + String.valueOf(true);
+
+            String url = "https://discover.mobileconnect.io/gsma/v2/discovery/?Redirect_URL=http%3A%2F%2Fjenkins" +
```

Comment:
Get this URL from a constant declared. Will make life easy for you when the URL changes :)